addons/mail/controllers/attachment.py

```diff
@@ -99,3 +100,30 @@ def mail_attachment_get_zip(self, file_ids, zip_name, **kw):
         ids_list = list(map(int, file_ids.split(',')))
         attachments = request.env['ir.attachment'].browse(ids_list)
         return self._make_zip(zip_name, attachments)
+
+    @http.route('/mail/attachment/first_page/<int:attachment_id>', methods=["GET"], type="http", auth="user", readonly=True)
+    def mail_attachment_pdf_first_page(self, attachment_id, access_token=None):
+        attachment = request.env['ir.attachment'].search([("id", "=", attachment_id)])
```

Comment:
There 2 are one of the rare case where we can/need to use browse rather than search as even if we cannot search the attachment we need to find it to call the check method or check the token.